timelock-agent/src/main/java/com/palantir/timelock/invariants/TimeLockActivityCheckerFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.invariants;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.palantir.atlasdb.config.ImmutableServerListConfig;
+import com.palantir.atlasdb.config.RemotingClientConfigs;
+import com.palantir.atlasdb.config.ServerListConfig;
+import com.palantir.atlasdb.factory.ServiceCreator;
+import com.palantir.atlasdb.util.MetricsManager;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.lock.v2.TimelockRpcClient;
+import com.palantir.timelock.config.TimeLockInstallConfiguration;
+import com.palantir.timelock.paxos.PaxosRemotingUtils;
+
+public class TimeLockActivityCheckerFactory {
+    private final TimeLockInstallConfiguration installConfiguration;
+    private final MetricsManager metricsManager;
+    private final UserAgent userAgent;
+
+    public TimeLockActivityCheckerFactory(
+            TimeLockInstallConfiguration installConfiguration, MetricsManager metricsManager, UserAgent userAgent) {
+        this.installConfiguration = installConfiguration;
+        this.metricsManager = metricsManager;
+        this.userAgent = userAgent;
+    }
+
+    public List<TimeLockActivityChecker> getTimeLockActivityCheckers() {
+        return installConfiguration.cluster()
```

Comment:
Ah, so the TACs are single node clients, which will throw NCLEs when we request a timestamp if they are not leaders, correct?